auth0/resource_auth0_tenant.go

```diff
@@ -0,0 +1,202 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/yieldr/go-auth0/management"
+)
+
+func newTenant() *schema.Resource {
+	return &schema.Resource{
+
+		Create: createTenant,
+		Read:   readTenant,
+		Update: updateTenant,
+		Delete: deleteTenant,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"change_password": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"html": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"guardian_mfa_page": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"html": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"default_audience": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"default_directory": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"error_page": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"html": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"show_log_link": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"url": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"friendly_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"picture_url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"allowed_logout_urls": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"session_lifetime": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"sandbox_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func createTenant(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	d.SetId(api.Domain)
```

Comment:
If this is the only place where `api.Domain` is used, do we really need it? Could we get away by generating a new id for it upon creation?

Reply:
That could work, but I wanted a way to enforce that only a single Tenant is ever added to a code base. If I use a random ID, then multiple tenants could be managed within the same code base with undesired behavior. I need to test what happens if you add two tenants with the domain as the ID.

Reply:
Good point, I haven't thought of that. But I suspect we'll have a problem regardless if two tenants are defined as the state will be mixed up.

Reply:
On second thought, relying on domain might enforce uniqueness (I now realize thats what you intended in the first place 🤦‍♂️. It's been a long day, sorry!)

In that case, if you can submit a PR with that change for yieldr/go-auth0 it would be awesome. Perhaps instead of a public property we could instead offer an accessor method? That way users won't change the domain after the client was initialized and authenticated.

Alternatively we could wrap `management.Management` from `provider.go` leaving the sdk as is. As the need for the domain is restricted to the tf provider, it would also be a nice alternative.

Reply:
I did a quick test, and it seems with either `api.Domain` and `resource.UniqueId()` the behavior is the same. The tenant is updated twice which will lead to confusion.

I propose we merge this as is, but warn against defining the tenant resource more than once until we figure out a way to enforce it with some kind of validation.